File: scripts/create_icon.py
#!/usr/bin/env python3
"""Generate Bible PWA app icons - bold '성경' on green background"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size, output_path):
    render_size = size * 4
    img = Image.new('RGBA', (render_size, render_size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)

    pad = render_size * 0.02
    corner = render_size * 0.18

    bg_color = (35, 100, 50)
    draw.rounded_rectangle([pad, pad, render_size - pad, render_size - pad], radius=corner, fill=bg_color)

    # "성경" - bold, white, centered
    font_size = int(render_size * 0.30)
    font_path = os.path.expanduser("~/.local/share/fonts/malgunbd.ttf")
    font = ImageFont.truetype(font_path, font_size)

    text = "성경"
    bbox = draw.textbbox((0, 0), text, font=font)
    tw = bbox[2] - bbox[0]
    th = bbox[3] - bbox[1]

    txt_img = Image.new('RGBA', (tw + 20, th + 20), (0, 0, 0, 0))
    txt_draw = ImageDraw.Draw(txt_img)
    txt_draw.text((-bbox[0] + 10, -bbox[1] + 10), text, fill=(255, 255, 255), font=font, stroke_width=6, stroke_fill=(255, 255, 255))

    # Paste centered on background (no stretch for Korean text)
    px = (render_size - txt_img.width) // 2
    py = (render_size - txt_img.height) // 2
    img.paste(txt_img, (px, py), txt_img)

    img = img.resize((size, size), Image.LANCZOS)
    img.save(output_path, 'PNG')
    print(f"Created {output_path} ({size}x{size})")


out_dir = os.path.join(os.path.dirname(__file__), '..', 'public')

create_icon(512, os.path.join(out_dir, 'icon-512.png'))
create_icon(192, os.path.join(out_dir, 'icon-192.png'))
create_icon(512, os.path.join(out_dir, 'icon.png'))
create_icon(512, os.path.join(out_dir, 'data', 'icon.png'))

File: scripts/create_icon.js
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

function createIcon(size) {
  const scale = 4;
  const renderSize = size * scale;
  const canvas = createCanvas(renderSize, renderSize);
  const ctx = canvas.getContext('2d');

  // Rounded rectangle background
  const pad = renderSize * 0.02;
  const corner = renderSize * 0.18;
  const bgColor = '#236432';

  ctx.beginPath();
  ctx.roundRect(pad, pad, renderSize - pad * 2, renderSize - pad * 2, corner);
  ctx.fillStyle = bgColor;
  ctx.fill();

  // Text settings - bold
  const fontSize = Math.floor(renderSize * 0.28);
  const fontStr = `900 ${fontSize}px Arial, sans-serif`;
  const strokeWidth = 6;
  const text = '성경';

  // Measure actual text bounds including descenders
  ctx.font = fontStr;
  const metrics = ctx.measureText(text);
  const actualTop = metrics.actualBoundingBoxAscent;
  const actualBottom = metrics.actualBoundingBoxDescent;
  const actualLeft = metrics.actualBoundingBoxLeft;
  const actualRight = metrics.actualBoundingBoxRight;
  const textWidth = actualLeft + actualRight + strokeWidth * 2;
  const textHeight = actualTop + actualBottom + strokeWidth * 2;

  // Draw text on temporary canvas with enough room for descenders
  const margin = 10;
  const txtCanvas = createCanvas(Math.ceil(textWidth) + margin * 2, Math.ceil(textHeight) + margin * 2);
  const txtCtx = txtCanvas.getContext('2d');
  txtCtx.font = fontStr;
  txtCtx.fillStyle = '#ffffff';
  txtCtx.strokeStyle = '#ffffff';
  txtCtx.lineWidth = strokeWidth;
  txtCtx.lineJoin = 'round';
  txtCtx.textAlign = 'left';
  txtCtx.textBaseline = 'alphabetic';

  // Position text so ascenders and descenders both fit
  const tx = actualLeft + strokeWidth + margin;
  const ty = actualTop + strokeWidth + margin;

  // Stroke first for extra thickness, then fill
  txtCtx.strokeText(text, tx, ty);
  txtCtx.fillText(text, tx, ty);

  // Draw text centered on main canvas (no stretch for Korean)
  const destWidth = txtCanvas.width;
  const destHeight = txtCanvas.height;
  const dx = (renderSize - destWidth) / 2;
  const dy = (renderSize - destHeight) / 2;

  ctx.drawImage(txtCanvas, 0, 0, txtCanvas.width, txtCanvas.height, dx, dy, destWidth, destHeight);

  // Downscale to target size
  const finalCanvas = createCanvas(size, size);
  const finalCtx = finalCanvas.getContext('2d');
  finalCtx.drawImage(canvas, 0, 0, renderSize, renderSize, 0, 0, size, size);

  return finalCanvas;
}

const outDir = path.join(__dirname, '..', 'public');

[512, 192].forEach(size => {
  const canvas = createIcon(size);
  const buffer = canvas.toBuffer('image/png');
  const filePath = path.join(outDir, `icon-${size}.png`);
  fs.writeFileSync(filePath, buffer);
  console.log(`Created ${filePath} (${size}x${size})`);
});

// Also create icon.png (same as 512)
const canvas512 = createIcon(512);
fs.writeFileSync(path.join(outDir, 'icon.png'), canvas512.toBuffer('image/png'));
console.log('Created icon.png (512x512)');

// Also copy to data/icon.png
fs.writeFileSync(path.join(outDir, 'data', 'icon.png'), canvas512.toBuffer('image/png'));
console.log('Created data/icon.png (512x512)');
